### ISSUE TITLE
Cloning new repo hangs

### ISSUE THREAD

**Issue by jrahaim** [fastlane/match#100]
>  match development --verbose

```
+----------------+----------------------------------------------------+
|                       Summary for match 0.2.5                       |
+----------------+----------------------------------------------------+
| git_url        | https://github.com/XXXX/iOSdevcertificates.git |
| type           | development                                        |
| app_identifier | com.XXXXXXXXX.ios                                   |
| username       | XXXXXXXXX                                           |
| keychain_name  | login.keychain                                     |
| readonly       | false                                              |
| team_id        | XXXXXXXXXXXX                                         |
| verbose        | true                                               |
| force          | false                                              |
| shallow_clone  | true                                               |
+----------------+----------------------------------------------------+

INFO [2016-02-02 10:49:06.61]: Cloning remote git repo...
INFO [2016-02-02 10:49:06.61]: $ git clone 'https://github.com/XXXXXXXX/iOSdevcertificates.git' '/var/folders/g1/0v_8w6bn2cdg8l_gvm_4yv4m0000gr/T/d20160202-44490-p8kb0n' --depth 1
INFO [2016-02-02 10:49:06.61]: ▸ Cloning into '/var/folders/g1/0v_8w6bn2cdg8l_gvm_4yv4m0000gr/T/d20160202-44490-p8kb0n'...
```

And it hangs there forever.

**Comment by KrauseFx** [comment]
Duplicate of https://github.com/fastlane/match/issues/40
What happens when you run the printed command manually?

**Comment by jrahaim** [comment]
It asks for a username/password. Since we use personal access tokens, I had to add the access tokenit to the git_URL above (I couldn't get it to read it from the Jenkins settings). This seems like a cludgy and dangerous work around though. Note: I am now 20+ hours into trying to get Jankins and fast lane set up and am really really frustrated with it.

**Comment by mfurtak** [comment]
@jrahaim When you specified the username and token in the git URL were you able to complete the operation through `match` without hanging?

**Comment by jmays86** [comment]
Same issue, none of the suggestions on #40 resolve the issue and when running the command manually, it complains about the folder in var already existing. I'm assuming thats because the git clone process may have started.

Xcode command line tools are installed and up to date and we can manually clone the repo outside of match.

**Comment by jmays86** [comment]
Just to add further information, we have match working on two machines, and not working on two others so if you need any debug information, I can provide that.

**Comment by jmays86** [comment]
Just an update for anybody reading this...

We initially skipped the match init as we have a lot of apps to manage and thought going straight to match and manually entering the various parameters should suffice. Turns out you need match init regardless and from that point onwards you can use match standalone without being in the project folder (by manually entering the details required) - which works well for us.

We then still periodically encountered the issue which was resolved by switching from the HTTPS URL bitbucket provided to SSH.

Working on all four machines now.

Thanks

**Comment by mfurtak** [comment]
@jmays86 Thanks for this info! :+1: 

It makes sense to me that the GitHub HTTPS git URL is causing the hang while prompting for credentials.

@jrahaim I believe that injecting the desired credentials into the git URL will work for your use case. Something like

```ruby
git_url = "https://#{ENV['GITHUB_USER']}:#{ENV['GITHUB_TOKEN']}@github.com/XXXX/iOSdevcertificates.git"
```

should get you unblocked if you can get those values set in your environment through Jenkins. I don't have much experience with Jenkins, but it seems like something like the [EnvInject Plugin](https://wiki.jenkins-ci.org/display/JENKINS/EnvInject+Plugin) may be needed.

We should certainly do more than just hang in this scenario, though, so I'll see what we can do about that!

**Comment by mfurtak** [comment]
@jrahaim We'll see if we can improve `match`'s behavior so that we don't get a hang with zero context provided!

**Comment by jeffreyjackson** [comment]
Hey i just seemed to have more luck with this.  For my CI machine, i had to end up using git@github.com:jeffreyjackson/certificates for the URL instead of the https:// or ssh+git://

**Comment by Sega-Zero** [comment]
Is there any way to pass credentials via separate command-line parameters for match?

**Comment by mfurtak** [comment]
@Sega-Zero That is not currently possible, no.

**Comment by Sega-Zero** [comment]
Well, then, at least stop log to console full git url with username/password? 
If I don't specify it - it's not working. If I do - log contains password which is very unacceptable IMHO

**Comment by mfurtak** [comment]
@Sega-Zero Great point. I would happily approve a PR that improves this! :+1:

**Comment by Sega-Zero** [comment]
I'm not familiar with ruby, unfortunately. Maybe, adding `:git_url` to `hide_keys`would be enough? Something like this
```ruby
FastlaneCore::PrintTable.print_values(config: params,
                                         hide_keys: [:workspace, :git_url],
                                             title: "Summary for match #{Match::VERSION}")
```

**Comment by KrauseFx** [comment]
@Sega-Zero This might work, however usually we want to the `git_url` to be shown. This should only be hidden if it contains the password. This should be possible to detect, as there are other special characters.

**Comment by Sega-Zero** [comment]
A better way would be showing `git_url` without credentials, I think. It's pretty easy to cut them off

**Comment by fastlanebot** [comment]
This issue was migrated to https://github.com/fastlane/fastlane/issues/3423. Please post all further comments there.

`fastlane` is now a mono repo, you can read more about the change in our [blog post](https://krausefx.com/blog/our-goal-to-unify-fastlane-tools). All tools are now available in the [fastlane main repo](https://github.com/fastlane/fastlane) :rocket:
